### DISCUSSION TITLE
How to set up an environment for the package to run?

### DISCUSSION THREAD

**Discussion by aleeusgr** [mlabs-haskell/plutip#140] [General]
Hello!
I want to learn (and create docs) on how to use plutip for writing tests; For this I think I should first write one, I found [this amazing doc](https://github.com/mlabs-haskell/plutip/blob/master/docs/interactive-plutip.md) here in the repo, but there is a gap in my knowledge.
I need to set up an environment for my test suite, my first choice was to do `cabal init` and then add plutip to the .cabal file. I assume this will allow me to import functions in my Main.hs, and use `cabal run` on my unit test.

However I get import errors, and plutip is not on Hackage. 
Therefore I need to import it some other way. 
What would you advice?

Since I use nixOS, I hope to build my program as a flake. 
I added
```
inputs = {
        plutip.url = "github:mlabs-haskell/plutip";
  }; 
```
and 
`outputs = { self, nixpkgs, flake-utils, plutip }:`

I understand that writing a flake involves several design choices, so I chose an [existing solution](https://github.com/mhwombat/hello-haskell-flake). 

Please help me figure out how to make the package available to import in app/Main.hs

Thank you!

**Comment by mikekeke** [comment]
Hi @aleeusgr! Great to see you here!
Here is an example of flake with Plutip: https://github.com/MitchyCola/plutip-flake

Example is based on the real project setup. Kudos to @MitchyCola for providing this template!❤️

**Reply by aleeusgr** [reply]
Absolutely!
So happy to see it!
Now my goal is to test it and write some tests with tasty for the project I'm working on right now.

**Reply by mikekeke** [reply]
If you will face any issues, please let us know!

**Comment by aleeusgr** [comment]
https://github.com/aleeusgr/plutus-testing

This a repo in style of Gimbalabs PPBL, an example of project based learning.

I will experiment there with plutip and testing in general.

The goal is to create a material to be used in onboarding new Devs and bringing them up to speed in Cardano development

**Reply by mikekeke** [reply]
Hey!
Great news!
Feel free to add the link and info into Plutips README if you wish ☺️
